documentation/modules/upgrading/con-upgrade-strategies-for-upgrading-clients.adoc

```diff
@@ -6,65 +6,43 @@
-
-This strategy is straightforward, and avoids any broker down-conversion.
-However, it assumes that all consumers in your organization can be upgraded in a coordinated way, and it does not work for applications that are both consumers and producers.
-There is also a risk that, if there is a problem with the upgraded clients, new-format messages might get added to the message log so that you cannot revert to the previous consumer version.
-
-.Topic-level consumers first strategy
-
-For each topic:
-
-. Upgrade all the consuming applications.
-. Change the topic-level `message.format.version` to the new version.
-. Upgrade all the producing applications.
-
-This strategy avoids any broker down-conversion, and means you can proceed on a topic-by-topic basis. It does not work for applications that are both consumers and producers of the same topic. Again, it has the risk that, if there is a problem with the upgraded clients, new-format messages might get added to the message log.
-
-.Topic-level consumers first strategy with down conversion
-
-For each topic:
+For the consumers first strategy with down-conversion at the topic level, you perform the following steps to upgrade clients:
 
 . Change the topic-level `message.format.version` to the old version
 (or rely on the topic defaulting to the broker-level `log.message.format.version`).
 . Upgrade all the consuming and producing applications.
 . Verify that the upgraded applications function correctly.
 . Change the topic-level `message.format.version` to the new version.
 
-This strategy requires broker down-conversion, but the load on the brokers is minimized because it is only required for a single topic (or small group of topics) at a time. It also works for applications that are both consumers and producers of the same topic. This approach ensures that the upgraded producers and consumers are working correctly before you commit to using the new message format version.
-
+Although this strategy uses broker down-conversion, the load on the brokers is minimized because it is only required for a single topic (or small group of topics) at a time. 
+It also works for applications that are both consumers and producers of the same topic. 
+The approach ensures that the upgraded producers and consumers are working correctly before you commit to using the new message format version.
 The main drawback of this approach is that it can be complicated to manage in a cluster with many topics and applications.
 
-NOTE: It is also possible to apply multiple strategies.
-For example, for the first few applications and topics the
-"per-topic consumers first, with down conversion" strategy can be used.
-When this has proved successful another, more efficient strategy can be considered acceptable to use instead.
+TIP: You can use Kafka metrics to monitor the performance of message down-conversion. 
+The _message conversion time_ metric measures the time taken to perform message conversion. 
```

Comment:
Let's include the MBean name or a link to the upstream doc.

Maybe it's also worth mentioning the metrics that track client names and versions:
`kafka.server:type=socket-server-metrics,clientSoftwareName=<name>,clientSoftwareVersion=<version>,listener=<listener>,networkProcessor=<processor>`

Reply:
Updated as follows:

````
[TIP]
====
The following Kafka broker metrics help monitor the performance of message down-conversion:

* `kafka.network:type=RequestMetrics,name=MessageConversionsTimeMs,request={Produce|Fetch}` provides metrics on the time taken to perform message conversion. 
* `kafka.server:type=BrokerTopicMetrics,name={Produce|Fetch}MessageConversionsPerSec,topic=([-.\w]+)` provides metrics on the number of messages converted over a period of time.  

You can track Kafka client names and versions using this metric:

* `kafka.server:type=socket-server-metrics,clientSoftwareName=<name>,clientSoftwareVersion=<version>,listener=<listener>,networkProcessor=<processor>`  
====
````